parser.go

```diff
@@ -158,24 +164,36 @@ func (p *Parser) parseEscapedExpansion() (EscapedExpansion, error) {
 	}
 }
 
-func (p *Parser) parseExpansion() (Expansion, error) {
+func (p *Parser) parseExpansion() (result Expansion, isLiteral bool, err error) {
 	if c := p.nextRune(); c != '$' {
-		return nil, fmt.Errorf("Expected expansion to start with $, got %c", c)
+		err = fmt.Errorf("Expected expansion to start with $, got %c", c)
+		return
```

Comment:
This function is just about too long for naked returns, IMO.

```suggestion
		return nil, false, fmt.Errorf("Expected expansion to start with $, got %c", c)
```